### ISSUE TITLE
$.pjax.reload reloads the whole page when url is specified and normalizeTrailingSlash is true

### ISSUE THREAD

**Issue by letstestcode** [yiisoft/yii2#16483]
Hello everybody.

I have configs like this
```
 'urlManager' => [
...
            'normalizer' => [
                'class' => 'yii\web\UrlNormalizer',
                'collapseSlashes' => true,
                'normalizeTrailingSlash' => true,
            ],
```
I have the GridView via Pjax on the page.
```    
    <?php Pjax::begin(['id' => 'any-pjax-container', 'timeout' => false]); ?>
    <?= GridView::widget([
        'dataProvider' => $dataProvider,
        'filterModel' => $searchModel,
... etc
```
Also I have $.pjax.reload fired on the some event.
```
// It reloads the whole page
$.pjax.reload("#any-pjax-container", {
    url: "/site/index?something=anything",
});
```
It reloads the whole page.

But if I pass the url with the trailing slash it works fine.
```
// It works fine
$.pjax.reload("#any-pjax-container", {
    url: "/site/index/?something=anything",
});
```
I have lost a lot of time to find the solution. Maybe it could be fixed in the framework.

**Comment by samdark** [comment]
Do you have any idea why that happens and how to fix it?

**Comment by letstestcode** [comment]
I think it because of pjax faces redirect when he tries to get content from the url. UrlNormalizer redirects request from /site/index?something=anything to /site/index/?something=anything

**Comment by rob006** [comment]
url: "/site/index?something=anything",

How did you get this URL? If your UrlManager is configured with trailing slash as suffix, you should get `/site/index/?something=anything`.

**Comment by letstestcode** [comment]
I typed it manually directly in my js. I didn't use Url::to or something.

**Comment by samdark** [comment]
Then it's not something that should be fixed on framework side.
